src/main/java/nl/mpcjanssen/simpletask/util/Util.kt

```diff
@@ -474,48 +475,77 @@ fun readAsset(assets: AssetManager, name: String): String {
     return buf.toString()
 }
 
-fun getRelativeThresholdDate(task: Task, ctx: Context): String? {
-    val date = task.thresholdDate
-    if (date != null) {
-        date.toDateTime()?.let {
-            return "T: " + RelativeDate.getRelativeDate(ctx, it)
-        }
-    }
-    return null
+fun getRelativeThresholdDate(task: Task, app: TodoApplication): String? {
+    val date = task.thresholdDate ?: return null
+    return getRelativeDate(app, "T: ", date).toString()
 }
 
-fun getRelativeDueDate(task: Task, application: TodoApplication, dueTodayColor: Int, overDueColor: Int, useColor: Boolean): SpannableString? {
-    val date = task.dueDate
-    if (date != null) {
-        date.toDateTime()?.let {
-            val relativeDate = RelativeDate.getRelativeDate(application, it)
-            val ss = SpannableString("Due: " + relativeDate)
-            if (date == application.today && useColor) {
-                setColor(ss, dueTodayColor)
-            } else if ((application.today.compareTo(date) > 0) && useColor) {
-                setColor(ss, overDueColor)
-            }
-            return ss
-        }
-    }
-    return null
+fun getRelativeDueDate(task: Task, app: TodoApplication): SpannableString? {
+    val date = task.dueDate ?: return null
+    return getRelativeDate(app, "Due: ", date)
 }
```

Comment:
See comment on `getRelativeThresholdDate` above. Also, moved color setting code inside the getRelativeDate function, which was the entire point of this endeavor -- so that the color and the text can be set using the same logic and therefore can never get out of sync.